File: TypeScript/Problem2.ts
function getMinPatches(nums: number[], n: number): number {
    let minPatches = 0;
    let miss = 1
    let i = 0;

    while (miss <= n) {
        if (i < nums.length && nums[i] <= miss) {
            miss += nums[i];
            i++;
        } else {
            miss += miss;
            minPatches++;
        }
    }

    return minPatches;
};

const nums1 = [1,3];
const n1 = 6;
const result1 = getMinPatches(nums1, n1);
console.log(result1);

const nums2 = [1,5,10];
const n2 = 20;
const result2 = getMinPatches(nums2, n2);
console.log(result2);

const nums3 = [1,2,2];
const n3 = 5;
const result3 = getMinPatches(nums3, n3);
console.log(result3);

File: python/Problem2.py
class Solution:
    def getMinPatches(self, nums: list[int], n: int) -> int:
        min_patches = 0
        miss = 1
        i = 0

        while (miss <= n):
            if(i < len(nums) and nums[i] <= miss):
                miss += nums[i]
                i += 1
            else:
                miss += miss
                min_patches += 1

        return min_patches

solution = Solution()

nums1 = [1,3]
n1 = 6
result1 = solution.getMinPatches(nums1, n1)
print(result1)

nums2 = [1,5,10]
n2 = 20
result2 = solution.getMinPatches(nums2, n2)
print(result2)

nums3 = [1,2,2]
n3 = 5
result3 = solution.getMinPatches(nums3, n3)
print(result3)